tesseract_visualization/include/tesseract_visualization/visualization.h

```diff
@@ -64,10 +60,10 @@ class Visualization
 
   /**
    * @brief Initialize the visualization tool
-   * @param env The environment object
+   * @detail It is recommend to check if connected before plotting any information
    * @return True if successful, otherwise false
    */
-  virtual bool init(tesseract_environment::Environment::ConstPtr env) = 0;
+  virtual bool init() = 0;
```

Comment:
Normally, I would say that having two-step initialization is not a great idea. In this case, I assume that visualizations might be plugins in the future, in which case the `init` function is useful for setting up the class. I would suggest providing some sort of configuration object to the `init` function (e.g. YAML node, XMLRPC value, XML document, etc.) so it's internal parameters can be set dynamically

Reply:
Yep, it is currently intended to support the plugin functionality.  Shouldn't the init be universal to all implementation for the purpose of plugins? The config object(YAML node, XMLRPC value, XML document, etc.) seems specific to the implementation which could be handled by the implementation class, would we want this in the base class?

Reply:
> Shouldn't the init be universal to all implementation for the purpose of plugins? The config object(YAML node, XMLRPC value, XML document, etc.) seems specific to the implementation which could be handled by the implementation class, would we want this in the base class?

Yes, the configuration object type should be generic and the same for all implementations, but its contents can change at runtime depending on how you need to configure the visualization. 

Normally you might provide implementation-specific parameters to these visualization implementations in their constructor. If they are plugins, though, they must be default constructable, so this isn't possible. It's valuable to have an alternative means of passing in parameters to visualization implementations in a generic way. This is where the `init` function with generic configuration object comes in. The configuration object should be a generic data type (like YAML node, XMLRPC value, `void*`, `std::any`, etc.) that is capable of dynamically representing different data, and the implementation should know how to interpret that generic data type to set up it's internal members.

In ROS-land, I typically use XMLRPC values because I can load them directly from the ROS parameter server and give them to a plugin. I would give the edge to YAML, XMLRPC, and XML over `void*` and `std::any` types because they are more straightforward to create from what we consider normal configuration files (e.g. text file, YAML file, XML file)

Reply:
We decided to go wit std::any.

Reply:
It would be beneficial if the `Marker` base didn't have two-step initialization; all setup should really be done in the constructor. I think the model below would be a better way of handling plugins such that they (the plugins) can be configured after instantiation, but the marker classes can be fully initialized using only their constructor

```c++
// Marker plugin base class (i.e. type returned by plugin loader)
class MarkerPlugin : public Marker
{
public:
  // Calls the internal marker base class virtual function
  virtual void display() override final
  {
    return marker_->display();
  }
  
  // Provides a means for creating a specific type of Marker class
  virtual void init(std::any params) = 0;

protected:
  // The actual marker that can create a display
  std::unique_ptr<Marker> marker_;
};

// Plugin implementation (only responsible for creating marker type from parameters)
class AxisMarkerPlugin : public MarkerPlugin
{
public:
  virtual void init(std::any params) override
  {
    AxisMarkerConfig axis_marker_config = std::any_cast<AxisMarkerConfig>(any);
    marker = std::make_unique<AxisMarkerType>(config_object);
  }
};
```
I understand we're not implementing plugins in this PR, but I don't think we should introduce the `init` function into any `Marker` classes in this PR

Reply:
@marip8 I removed the init method.

Reply:
I held off with creating the plugin class until we get around to discussing how to handle loading plugins with Tesseract.